src/files/FilesPage.js

```diff
@@ -357,7 +357,11 @@ const FilesPage = ({
 
       <MainView t={t} files={files} remotePins={remotePins} pendingPins={pendingPins} failedPins={failedPins} doExploreUserProvidedPath={doExploreUserProvidedPath}/>
 
-      <Preview files={files} onDownload={() => onDownload([files])} />
+      <Preview files={files} onDownload={() => onDownload([files])} onClose={() => {
+        // Navigate to parent directory to close file view
+        const parentPath = files.path.substring(0, files.path.lastIndexOf('/')) || '/'
+        doUpdateHash(parentPath)
+      }} />
```

Comment:
there should be an easier way to get parentPath.. we are parsing the paths in a number of places. I looked around and it doesn't seem like that's piped through to the files state. I'll push up a quick fix.

Reply:
fixed by [`9c6c697` (#2401)](https://github.com/ipfs/ipfs-webui/pull/2401/commits/9c6c6973610cc406525b11fec2357f7c348c8419)